### ISSUE TITLE
Investigate Savepoints not being released by OS process and causing memory leak

### ISSUE THREAD

**Issue by raphjaph** [ordinals/ord#2439]

**Comment by raphjaph** [comment]
Yes, this probably still is a problem. It's hard to reproduce and only occurs on some systems and configurations. Is your Bitcoin Core node fully synced?

**Comment by radicleart** [comment]
Yep - bitcoin synced on mainnet.

I've managed to work around by grabbing a snapshot of the index [here](https://ordstuff.info/indexes/) the server synced from there.

**Comment by JD-Wang** [comment]
hi,  Have you solved this problem? My 8c16g server is the same. Ord occupies 100% of the memory and is then killed. After a few days, there is still no way to solve it

**Comment by endangurura** [comment]
@raphjaph This is still an issue. Has anyone found a fix?

**Comment by raphjaph** [comment]
It was a bug in our database (redb), was fixed here https://github.com/cberner/redb/pull/843 and here https://github.com/cberner/redb/pull/841. We did a hot fix in the current version of ord though. Which version are you using?

**Comment by raphjaph** [comment]
I just merged #4169 and I think this fixed it. Please reopen if not
